.github/workflows/vale-tdbx.yml

```diff
@@ -0,0 +1,38 @@
+name: vale-checks
+on:
+  pull_request:
+    paths:
+      - "source/**"
+
+jobs:
+  vale:
+    name: TDBX Vale rules
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@master
+
+      - id: files
+        uses: masesgroup/retrieve-changed-files@v2
+        with:
+          format: 'csv'
+
+      - name: checkout-latest-rules
+        uses: actions/checkout@master
+        with:
+          repository: mongodb/mongodb-vale-action
+          path: './tdbx-vale-rules'
+          token: ${{secrets.GITHUB_TOKEN}}
+
+      - name: move-files-for-vale-action
+        run: |
+            cp tdbx-vale-rules/.vale.ini .vale.ini
+            mkdir -p .github/styles/
+            cp -rf tdbx-vale-rules/.github/styles/ .github/
+      - name: run-vale
```

Comment:
Nit/C: Unsure if you need a line between the:

        run: |
            cp tdbx-vale-rules/.vale.ini .vale.ini
            mkdir -p .github/styles/
            cp -rf tdbx-vale-rules/.github/styles/ .github/

and the name: run-vale, as the other name/run pairs all have spaces/line breaks between each name/run group